frontend/src/features/admin-form/create/builder-and-design/BuilderAndDesignDrawer/FieldListDrawer/field-panels/PaymentsInputPanel/PaymentsInputPanel.tsx

```diff
@@ -196,63 +160,99 @@ export const PaymentInput = ({ isDisabled }: { isDisabled: boolean }) => {
         <Toggle
           {...register('enabled', {
             // Retrigger validation to remove errors when payment is toggled from enabled -> disabled
-            onChange: () => paymentIsEnabled && trigger(),
+            onChange: () => isUsingPayment && trigger(),
           })}
           description="Payment field will not be shown when this is toggled off. Respondents can still submit the form."
           label="Enable payment"
         />
       </FormControl>
-
-      <FormControl
-        isReadOnly={paymentsMutation.isLoading}
-        isInvalid={!!errors.description}
-        isDisabled={!paymentIsEnabled}
-        isRequired
-      >
-        <FormLabel description="This will be reflected on the payment invoice">
-          Product/service name
-        </FormLabel>
-        <Input
-          placeholder="Product/service name"
-          {...register('description', {
-            required: paymentIsEnabled && 'Please enter a payment description',
-          })}
-        />
-        <FormErrorMessage>{errors.description?.message}</FormErrorMessage>
-      </FormControl>
-
-      <FormControl
-        isReadOnly={paymentsMutation.isLoading}
-        isInvalid={!!errors.display_amount}
-        isDisabled={!paymentIsEnabled}
-        isRequired
-      >
-        <FormLabel isRequired description="Amount should include GST">
-          Payment amount
-        </FormLabel>
-        <Controller
-          name="display_amount"
-          control={control}
-          rules={amountValidation}
-          render={({ field }) => (
-            <MoneyInput
-              flex={1}
-              step={0}
-              inputMode="decimal"
-              placeholder="0.00"
-              {...field}
-            />
-          )}
-        />
-        <FormErrorMessage>{errors.display_amount?.message}</FormErrorMessage>
-        {Number(clonedWatchedInputs.display_amount) > 1000 ? (
-          <InlineMessage variant="warning" mt="2rem" useMarkdown>
-            You would need to issue your own invoice for amounts above S$1000.
-            [Learn more about
-            this](https://guide.form.gov.sg/faq/faq/payments#simplified-tax-invoices-versus-regular-tax-invoices)
-          </InlineMessage>
-        ) : null}
-      </FormControl>
+      {isUsingPayment && (
```

Comment:
Good question. I had initially done that as well, but it was due to how `React.Fragment` works together with component's `children`.

In the parent component, `<CreatePageDrawerContentContainer />`, it checks for it's child to determine whether there should be a `<Divider />` inserted.

If we are encompass the whole payments block with a singular `isUsingPayment`, we would need to wrap the 4 child components under a singular `< />` 
```
<CreatePageDrawerContentContainer>
 {isUsingPayment && <>{...fourChildComponents}
</CreatePageDrawerContentContainer>
```

This will result in no `<Divider />` inserted between its children.

Reply:
Oh yeah... Hmm okay this should be transient issue right? Because of FRM-1016

After that the components should always be there less the switching between fixed and variable

Reply:
Yeah. Didn't want to really spend too much time writing a solution that will be eventually resolved xD.